charon/src/types.rs

```diff
@@ -371,6 +371,9 @@ pub struct Variant {
     pub meta: Meta,
     pub name: String,
     pub fields: FieldId::Vector<Field>,
+    #[serde(skip)]
+    /// The discriminant used at runtime.
```

Comment:
Can you add more information in this comment, to explain that we only need this in remove_read_discriminant to properly match the integer discriminant values to the variant ids when reconstructing the matches?